### PR TITLE
create and connect locations card in search results

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
Create a card renderer for the results of the locations search. Matching figma mockup. [TRM-25595](https://www.ebi.ac.uk/panda/jira/browse/TRM-25595)

## Approach
Create a card renderer, connect it to the results mega-component.

## Testing
Snapshot testing + interaction testing (checkbox selection and card click).
Also added tests on interaction testing on UniParc card renderer in order to improve the coverage

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniparc/components/results/UniParcCard.tsx [hunk 1]
```diff
@@ -76,13 +76,13 @@ const UniRefCard: FC<{
...
-              <strong>Organism{organismCount === 1 ? '' : 's'}: </strong>
+              <strong>Organism{organismCount === 1 ? '' : 's'}:</strong>{' '}
               <LongNumber>{organismCount}</LongNumber>
             </span>
             <span className="result-card__info-bit">
-              <strong>UniprotKB entries:</strong> {uniProtKBCount.reviewed}{' '}
-              reviewed and <LongNumber>{uniProtKBCount.unreviewed}</LongNumber>{' '}
-              unreviewed
+              <strong>UniprotKB entries:</strong>{' '}
+              <LongNumber>{uniProtKBCount.reviewed}</LongNumber> reviewed and{' '}
```
Comment (xwatkins) [inline src/uniparc/components/results/UniParcCard.tsx:84]: Something for later (was planning on doing it last sprint but ran out of time) but I think we should use the `Card` links prop for these (we'll have to do with all namespaces at the same time).

File: src/supporting-data/locations/components/results/LocationsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              onChange={() => handleEntrySelection(data.id)}
+              data-testid="up-card-checkbox"
+            />
+          </div>
+        )}
+        <div className="result-card__right">
+          <h5>{data.name}</h5>
+          <div className="result-card__info-container">{data.definition}</div>
+          <div className="result-card__info-container">
+            <strong>Category:</strong> {data.category}
```
Comment (xwatkins) [inline src/supporting-data/locations/components/results/LocationsCard.tsx:46]: I know there's only 1, but think we should be using `RenderColumnInCard` whenever we can.

**Review by xwatkins — CHANGES_REQUESTED** [review]
small change

File: src/supporting-data/locations/components/results/LocationsCard.tsx (hunk 2 above)
Comment (aurel-l) [reply src/supporting-data/locations/components/results/LocationsCard.tsx:46]: See https://github.com/ebi-uniprot/uniprot-website/pull/319

Comment (aurel-l) [reply src/supporting-data/locations/components/results/LocationsCard.tsx:46]: See this https://github.com/ebi-uniprot/uniprot-website/blob/417d7b035c06a1fa437515c49f72b2d81f3611b8/src/supporting-data/locations/components/results/LocationsCard.tsx#L54 more specifically

Comment (xwatkins) [reply src/supporting-data/locations/components/results/LocationsCard.tsx:46]: Aaaaah

**Review by xwatkins — APPROVED** [review]

File: src/supporting-data/locations/components/results/LocationsCard.tsx (hunk 2 above)
Comment (aurel-l) [reply src/supporting-data/locations/components/results/LocationsCard.tsx:46]: This is specifically this change that made me enter a rabbit hole of changes that led to that other PR

**Pull request merged**